src/hotspot/share/code/nmethod.hpp

```diff
@@ -471,77 +471,75 @@ class nmethod : public CodeBlob {
   void oops_do_set_strong_done(nmethod* old_head);
 
 public:
-  enum class ChangeReason : u1 {
-    C1_codepatch,
-    C1_deoptimize,
-    C1_deoptimize_for_patching,
-    C1_predicate_failed_trap,
-    CI_replay,
-    JVMCI_invalidate_nmethod,
-    JVMCI_invalidate_nmethod_mirror,
-    JVMCI_materialize_virtual_object,
-    JVMCI_new_installation,
-    JVMCI_register_method,
-    JVMCI_replacing_with_new_code,
-    JVMCI_reprofile,
-    marked_for_deoptimization,
-    missing_exception_handler,
-    not_used,
-    OSR_invalidation_back_branch,
-    OSR_invalidation_for_compiling_with_C1,
-    OSR_invalidation_of_lower_level,
-    set_native_function,
-    uncommon_trap,
-    whitebox_deoptimization,
-    zombie,
+  // If you change anything in this enum please patch
+  // vmStructs_jvmci.cpp accordingly.
+  enum class InvalidationReason : s1 {
+    NOT_INVALIDATED = -1,
+    UNKNOWN,
+    C1_CODEPATCH,
+    C1_DEOPTIMIZE,
+    C1_DEOPTIMIZE_FOR_PATCHING,
+    C1_PREDICATE_FAILED_TRAP,
+    CI_REPLAY,
+    UNLOADING,
+    UNLOADING_COLD,
+    JVMCI_INVALIDATE,
+    JVMCI_MATERIALIZE_VIRTUAL_OBJECT,
+    JVMCI_REPLACED_WITH_NEW_CODE,
+    JVMCI_REPROFILE,
+    MARKED_FOR_DEOPTIMIZATION,
+    MISSING_EXCEPTION_HANDLER,
+    NOT_USED,
+    OSR_INVALIDATION_BACK_BRANCH,
+    OSR_INVALIDATION_FOR_COMPILING_WITH_C1,
+    OSR_INVALIDATION_OF_LOWER_LEVEL,
+    SET_NATIVE_FUNCTION,
+    UNCOMMON_TRAP,
+    WHITEBOX_DEOPTIMIZATION,
+    ZOMBIE,
+    LAST_REASON
```

Comment:
This isn't really the last reason, since it's not actually a reason.  So either `LAST_REASON = ZOMBIE` with adjustments to the range check or maybe `REASON_COUNT`?  Why do we need `UNKNOWN` since it seems unused?

Reply:
I'll looked other enums in HotSpot and there was a mixed of `*_COUNT`, `LAST_*`, etc. I opted to use `LAST_*` for no particular 'reason'. I'll change to `REASON_COUNT` as you suggest.